static/app/utils/replays/timer.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * EventTarget has ~97% browser support
+ */
+export class Timer extends EventTarget {
+  private _id: number | null = null;
+  private _active: boolean = false;
+  private _start: number = 0;
+  private _time: number = 0;
+  private _additionalTime: number = 0;
+  private _callbacks: [offset: number, callback: () => void][] = [];
+
+  constructor() {
+    super();
+  }
+
+  step = () => {
+    if (!this._active) {
+      return;
+    }
+
+    this._time = window.performance.now() - this._start;
+    for (let i = 0; i < this._callbacks.length; i++) {
+      if (this._time >= this._callbacks[i][0]) {
+        this._callbacks[i][1]();
+        // Remove from _callbacks
+        this._callbacks.splice(i, 1);
+      }
+    }
```

Comment:
Do we skip callbacks in the case where two in a row should run?

ie: the array is something like: `[t=1, t=2, t=3]` and the current `_time=2`
In this case we start with array.length==3 iterate to `i=0` and the callback for t=1 will run because `t=1` is less than `_time=2`
splice reduces the array to length==2, and i is set to 1, which points to `t=3`. t=2 is skipped :(
Then the array

Reply:
Ah yeah, I'll clean this up and add a test.

Reply:
Fixed here https://github.com/getsentry/sentry/pull/66111/commits/4a1bb5396c10670132b4f0b87f522bdcf51a64e9